crates/nostr/src/prelude.rs

```diff
@@ -5,41 +5,40 @@
 
 // External crates
 #[cfg(feature = "nip19")]
-pub use bech32::*;
+pub use bech32;
 #[cfg(feature = "nip06")]
-pub use bip39::*;
-#[cfg(feature = "nip06")]
-pub use bitcoin::*;
-pub use bitcoin_hashes::*;
-pub use secp256k1::*;
-pub use serde_json::*;
+pub use bip39;
+pub use secp256k1;
 pub use url::*;
 
 // Internal modules
-pub use crate::event::builder::*;
-pub use crate::event::id::*;
-pub use crate::event::kind::*;
-pub use crate::event::tag::*;
-pub use crate::event::unsigned::*;
-pub use crate::event::*;
-pub use crate::key::*;
-pub use crate::message::*;
-pub use crate::types::*;
-pub use crate::{Result, SECP256K1};
+pub use crate::event;
+pub use crate::event::builder;
+pub use crate::event::id;
+pub use crate::event::kind;
+pub use crate::event::tag;
+pub use crate::event::unsigned;
+pub use crate::key;
+pub use crate::message;
+pub use crate::types;
+pub use crate::Result;
+
+#[cfg(feature = "std")]
+pub use crate::SECP256K1;
 
 // NIPs
 #[cfg(feature = "nip04")]
-pub use crate::nips::nip04::*;
+pub use crate::nips::nip04;
 #[cfg(feature = "nip05")]
-pub use crate::nips::nip05::*;
+pub use crate::nips::nip05;
 #[cfg(feature = "nip06")]
-pub use crate::nips::nip06::*;
+pub use crate::nips::nip06;
 #[cfg(feature = "nip11")]
-pub use crate::nips::nip11::*;
-pub use crate::nips::nip13::*;
+pub use crate::nips::nip11;
+pub use crate::nips::nip13;
 #[cfg(feature = "nip19")]
-pub use crate::nips::nip19::*;
-pub use crate::nips::nip26::*;
+pub use crate::nips::nip19;
+pub use crate::nips::nip26;
 #[cfg(feature = "nip46")]
-pub use crate::nips::nip46::*;
-pub use crate::nips::nip65::*;
+pub use crate::nips::nip46;
+pub use crate::nips::nip65;
```

Comment:
I am honestly not sure, because it gives me a warning:

```
 --> nostr/crates/nostr/src/prelude.rs:14:9
   |
14 | pub use bitcoin_hashes::*;
   |         ^^^^^^^^^^^^^^^^^ the name `error` in the type namespace is first re-exported here
15 | pub use secp256k1::*;
16 | pub use serde_json::*;
   |         ------------- but the name `error` in the type namespace is also re-exported here
   |
   = note: `#[warn(ambiguous_glob_reexports)]` on by default
```

I think it is most likely because an `Error` named enum is declared in each module.

Reply:
For now, add `#![allow(ambiguous_glob_reexports)]` in the prelude module.

```rust
//! Prelude

#![allow(ambiguous_glob_reexports)]

// External crates
...
```